src/wallet/orchard.h

```diff
@@ -65,12 +65,40 @@ class OrchardWallet
     OrchardWallet(const OrchardWallet&) = delete;
     OrchardWallet& operator=(const OrchardWallet&) = delete;
 
-    void CheckpointNoteCommitmentTree() {
-        orchard_wallet_checkpoint(inner.get());
+    /**
+     * Reset the state of the wallet to be suitable for rescan from the NU5 activation
+     * height.  This removes all witness and spentness information from the wallet. The
+     * keystore is unmodified and decrypted note, nullifier, and conflict data are left
+     * in place with the expectation that they will be overwritten and/or updated in the
+     * rescan process.
```

Comment:
Is it possible for rescan to fail in such a way that the stale data is left in place even though the wallet has been reset?

Reply:
That will really depend upon wallet persistence, I think; this operation only removes in-memory data. I'll take the question into consideration as I implement the persistence operations though.